core/src/com/google/inject/internal/DefaultConstructionProxyFactory.java

```diff
@@ -46,16 +46,12 @@
 
     /*if[AOP]*/
     try {
-      net.sf.cglib.reflect.FastClass fc = BytecodeGen.newFastClassForMember(constructor);
-      if (fc != null) {
-        int index = fc.getIndex(constructor.getParameterTypes());
-        // We could just fall back to reflection in this case but I believe this should actually
-        // be impossible.
-        Preconditions.checkArgument(
-            index >= 0, "Could not find constructor %s in fast class", constructor);
-        return new FastClassProxy<T>(injectionPoint, constructor, fc, index);
+      BiFunction<Object, Object[], Object> fastConstructor =
+          BytecodeGen.fastConstructor(constructor);
+      if (fastConstructor != null) {
+        return new FastClassProxy<T>(injectionPoint, constructor, fastConstructor);
       }
-    } catch (net.sf.cglib.core.CodeGenerationException e) {
+    } catch (Exception | LinkageError e) {
```

Comment:
do we have any narrower type we can catch here, to avoid over-catching exceptions & suppressing real errors?

(same elsewhere we catch Exception or Throwable)

Reply:
This is meant to catch any exceptions that occurred while creating the fast constructor (not invocation as that happens inside `FastClassProxy`) and fall-through to use JDK reflection instead. CGLIB wrapped anything extending `Exception` inside `CodeGenerationException` so this just widens it to include linkage errors which can sometimes be thrown during class-loading. We'd want to fall back to JDK reflection in that case too.

We could consider logging the exception, but we'd still want to fall back to JDK reflection when we can't create the fast constructor.